vendor/bat-native-confirmations/src/bat/confirmations/internal/payments.cc

```diff
@@ -199,30 +294,48 @@ bool Payments::GetTransactionCountFromDictionary(
   return true;
 }
 
-bool Payments::HasPendingBalanceForLastMonth() const {
-  if (payments_.size() < 2) {
-    return false;
-  }
-
-  PaymentInfo payment(payments_.at(1));
+bool Payments::HasPendingBalanceForTransactionMonth(
+    const std::string& month) const {
+  auto payment = GetPaymentForTransactionMonth(month);
   if (payment.balance == 0.0) {
     return false;
   }
 
   return true;
 }
 
-bool Payments::HasPendingBalanceForThisMonth() const {
-  if (payments_.empty()) {
-    return false;
+PaymentInfo Payments::GetPaymentForTransactionMonth(
+    const std::string& month) const {
+  for (const auto& payment : payments_) {
+    if (payment.month == month) {
+      return payment;
+    }
   }
 
-  PaymentInfo payment(payments_.front());
-  if (payment.balance == 0.0) {
-    return false;
+  return PaymentInfo();
+}
+
+std::string Payments::GetTransactionMonth(const base::Time& time) const {
+  base::Time::Exploded time_exploded;
+  time.LocalExplode(&time_exploded);
+
+  return base::StringPrintf("%04d-%02d", time_exploded.year,
+      time_exploded.month);
+}
+
+std::string Payments::GetPreviousTransactionMonth(
+    const base::Time& time) const {
+  base::Time::Exploded time_exploded;
+  time.LocalExplode(&time_exploded);
+
+  time_exploded.month--;
+  if (time_exploded.month < 1) {
+    time_exploded.month += 12;
```

Comment:
I think `time_exploded.month = 12` makes the intent a little clearer here, even though it's the overall same effect.

Reply:
👍 fixed